src/checks/eslint.ts

```diff
@@ -1,35 +1,36 @@
+import { ESLint } from 'eslint';
 import { execaCommand } from 'execa';
-import normalizeNewline from '../util/normalizeNewline';
+
+const errorKey = 'errorCount' as const;
 
 export const title = 'ESLint';
 
 export default async function eslintCheck() {
   try {
-    await execaCommand('pnpm eslint . --max-warnings 0  --format compact');
+    await execaCommand('pnpm eslint . --max-warnings 0  --format json');
   } catch (error) {
     const { stdout } = error as { stdout: string };
-    const lines = stdout.split('\n');
 
     // If no ESLint problems detected, throw the error
-    if (!/^\d+ problems?$/.test(lines[lines.length - 2]!)) {
+    if (!new RegExp(`"${errorKey}":`).test(stdout)) {
```

Comment:
this is wrong, it won't fail on warnings

Reply:
778935c

following a similar approach as https://github.com/upleveled/eslint-config-upleveled/blob/99c14d946b1f52f7d37a32a3b4ac83b6aa3fcb0b/bin/install.js#L20

Reply:
hmm this is not true
https://github.com/upleveled/preflight/pull/377#discussion_r1173570338
the property is allways there
[11:53](https://upleveled.slack.com/archives/D017DL7RRU2/p1682070838770789)
interface LintResult {
        filePath: string;
        messages: Linter.LintMessage[];
        suppressedMessages: Linter.SuppressedLintMessage[];
        errorCount: number;
        fatalErrorCount: number;
        warningCount: number;
        fixableErrorCount: number;
        fixableWarningCount: number;
        output?: string | undefined;
        source?: string | undefined;
        usedDeprecatedRules: DeprecatedRuleUse[];
    }
[11:54](https://upleveled.slack.com/archives/D017DL7RRU2/p1682070850428899)
even if its errorCount:0

Reply:
instead of checking for two dynamic keys, look at your data when you have no errors, maybe you can see a better way of checking it